components/ide/code/extensionhelper/main.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-protocol"
+)
+
+const (
+	GITPOD_REPO_ROOT             = "GITPOD_REPO_ROOT"
+	GITPOD_WORKSPACE_CONTEXT_URL = "GITPOD_WORKSPACE_CONTEXT_URL"
+)
+
+func main() {
+	output := "--start-server"
+	extensions, _ := getExtensions()
+	for _, ext := range extensions {
+		output += " --install-extension " + ext
+	}
+	fmt.Print(output)
+}
+
+func getExtensions() (extensions []string, err error) {
+	workspaceContextUrl := os.Getenv(GITPOD_WORKSPACE_CONTEXT_URL)
+	if workspaceContextUrl != "" && strings.Contains(workspaceContextUrl, "github.com") {
+		extensions = append(extensions, "github.vscode-pull-request-github")
+	}
+	repoRoot := os.Getenv(GITPOD_REPO_ROOT)
+	if repoRoot == "" {
+		return
+	}
+	data, err := os.ReadFile(filepath.Join(repoRoot, ".gitpod.yml"))
```

Comment:
please add await here for content ready of supervisor endpoint otherwise it does not guarantee to be present on the disk

I guess it was a bug in VS Code Server customizations before, taht we don't check for it